lua/lspconfig/ui/lspinfo.lua

```diff
@@ -64,9 +64,12 @@ local function make_config_info(config, bufnr)
   local buffer_dir = api.nvim_buf_call(bufnr, function()
     return vim.fn.expand '%:p:h'
   end)
-  local root_dir = config.get_root_dir and config.get_root_dir(buffer_dir)
-  if root_dir then
-    config_info.root_dir = root_dir
+
+  if config.get_root_dir then
+    config_info.root_dir = coroutine.resume(coroutine.create(vim.schedul_wrap(function()
+      local root_dir = config.get_root_dir(buffer_dir)
+      coroutine.yield(root_dir)
+    end)))
```

Comment:
If `get_root_dir` calls `coroutine.yield` first, the result of that will be returned from this `coroutine.resume` (and as the second value, the first value is the status of the coroutine) and assigned to `root_dir`, which wouldn't be correct. We specifically want the result of `config.get_root_dir`.

Also there is a typo in `schedul_wrap`

Reply:
yep first value is status. a mistake. but i really don't like this in lspinfo ..

Reply:
how about now